extensions/cornerstone-3d/src/commandsModule.js

```diff
@@ -304,11 +322,146 @@ const commandsModule = ({ servicesManager }) => {
       const { viewport } = enabledElement;
       const options = { delta: direction };
 
-      cornerstone3DTools.utilities.stackScrollTool.scrollThroughStack(
+      cornerstoneTools.utilities.stackScrollTool.scrollThroughStack(
         viewport,
         options
       );
     },
+    async createSegmentationForDisplaySet({ displaySetInstanceUID }) {
+      const volumeId = displaySetInstanceUID;
+
+      const segmentationUID = cornerstone.utilities.uuidv4();
+      const segmentationId = `${volumeId}::${segmentationUID}`;
+
+      await cornerstone.volumeLoader.createAndCacheDerivedVolume(volumeId, {
+        volumeId: segmentationId,
+      });
+
+      // Add the segmentations to state
+      cornerstoneTools.segmentation.addSegmentations([
+        {
+          segmentationId,
+          representation: {
+            // The type of segmentation
+            type: cornerstoneTools.Enums.SegmentationRepresentations.Labelmap,
+            // The actual segmentation data, in the case of labelmap this is a
+            // reference to the source volume of the segmentation.
+            data: {
+              volumeId: segmentationId,
+            },
+          },
+        },
+      ]);
+
+      return segmentationId;
+    },
+    async addSegmentationRepresentationToToolGroup({
+      segmentationId,
+      toolGroupId,
+      options,
+    }) {
+      const { representationType } = options;
+
+      // // Add the segmentation representation to the toolgroup
+      await cornerstoneTools.segmentation.addSegmentationRepresentations(
+        toolGroupId,
+        [
+          {
+            segmentationId,
+            type: representationType,
+          },
+        ]
+      );
+    },
+    getLabelmapVolumes: ({ segmentations }) => {
+      if (!segmentations || !segmentations.length) {
+        segmentations = SegmentationService.getSegmentations();
+      }
+
+      const labelmapVolumes = segmentations.map(segmentation => {
+        return cornerstone.cache.getVolume(segmentation.id);
+      });
+
+      return labelmapVolumes;
+    },
+    getSegmentationReport: ({ segmentations }) => {
```

Comment:
This seems extremely specific to whatever the tmtv mode wants and seems like its a bit strange at the cornerstone extension level. Can this be hoisted to the tmtv extension?

Reply:
getCSVReport